diff --git a/libs/math/config/has_long_double_support.cpp b/libs/math/config/has_long_double_support.cpp
index d314cf3..9022408 100644
--- a/libs/math/config/has_long_double_support.cpp
+++ b/libs/math/config/has_long_double_support.cpp
@@ -8,3 +8,7 @@
 #ifdef BOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS
 #error "long double support is not supported by Boost.Math on this Plaform: the long double version of the TR1 library will not be built."
 #endif
+
+int main(int argc, char *argv[]) {
+  return 0;
+}
