Author: Hesham <hesham.essam.mail@gmail.com>
Date:   Tue Sep 13 00:09:57 2022 +0200

    Change rpath of glib libraries

diff --git a/gio/meson.build b/gio/meson.build
index ca96930..ec2a037 100644
--- a/gio/meson.build
+++ b/gio/meson.build
@@ -820,6 +820,7 @@ libgio = library('gio-2.0',
   objc_args : gio_c_args,
   # intl.lib is not compatible with SAFESEH
   link_args : [noseh_link_args, glib_link_flags],
+  install_rpath: '$ORIGIN',
 )
 
 if get_option('gio_module_dir') != ''
diff --git a/glib/meson.build b/glib/meson.build
index 6062c11..0dd4808 100644
--- a/glib/meson.build
+++ b/glib/meson.build
@@ -378,6 +378,7 @@ libglib = library('glib-2.0',
   dependencies : [pcre2, thread_dep, librt] + libintl_deps + libiconv + platform_deps + [gnulib_libm_dependency, libm] + [libsysprof_capture_dep],
   c_args : glib_c_args,
   objc_args : glib_c_args,
+  install_rpath: '$ORIGIN',
 )
 
 libglib_dep = declare_dependency(
diff --git a/gmodule/meson.build b/gmodule/meson.build
index 0c5aad3..58e1143 100644
--- a/gmodule/meson.build
+++ b/gmodule/meson.build
@@ -87,6 +87,7 @@ libgmodule = library('gmodule-2.0',
   dependencies : [libdl_dep, libglib_dep],
   c_args : ['-DG_LOG_DOMAIN="GModule"'] + glib_hidden_visibility_args,
   link_args : [glib_link_flags],
+  install_rpath: '$ORIGIN',
 )
 
 supported_var = 'gmodule_supported=@0@'.format(g_module_impl != '0')
diff --git a/gobject/meson.build b/gobject/meson.build
index 07d096f..00f4ef0 100644
--- a/gobject/meson.build
+++ b/gobject/meson.build
@@ -136,6 +136,7 @@ libgobject = library('gobject-2.0',
   dependencies : [libffi_dep, libglib_dep],
   c_args : ['-DG_LOG_DOMAIN="GLib-GObject"', '-DGOBJECT_COMPILATION'] + glib_hidden_visibility_args,
   link_args : glib_link_flags,
+  install_rpath: '$ORIGIN',
 )
 
 pkg.generate(libgobject,
diff --git a/gthread/meson.build b/gthread/meson.build
index 94a9ce5..8a252ed 100644
--- a/gthread/meson.build
+++ b/gthread/meson.build
@@ -21,6 +21,7 @@ libgthread = library('gthread-2.0',
   dependencies : [libglib_dep],
   c_args : ['-DG_LOG_DOMAIN="GThread"' ] + glib_hidden_visibility_args,
   link_args : glib_link_flags,
+  install_rpath: '$ORIGIN',
 )
 
 pkg.generate(libgthread,
