diff --git a/a/CMakeLists.txt b/b/CMakeLists.txt
index 3259ca9..8168bb4 100644
--- a/a/CMakeLists.txt
+++ b/b/CMakeLists.txt
@@ -247,19 +247,19 @@ if (HB_HAVE_ICU)
   add_definitions(-DHAVE_ICU)
 
   # https://github.com/WebKit/webkit/blob/fdd7733f2f30eab7fe096a9791f98c60f62f49c0/Source/cmake/FindICU.cmake
-  find_package(PkgConfig)
-  pkg_check_modules(PC_ICU QUIET icu-uc)
+  # find_package(PkgConfig)
+  # pkg_check_modules(PC_ICU QUIET icu-uc)
 
-  find_path(ICU_INCLUDE_DIR NAMES unicode/utypes.h HINTS ${PC_ICU_INCLUDE_DIRS} ${PC_ICU_INCLUDEDIR})
-  find_library(ICU_LIBRARY NAMES libicuuc cygicuuc cygicuuc32 icuuc HINTS ${PC_ICU_LIBRARY_DIRS} ${PC_ICU_LIBDIR})
+  # find_path(ICU_INCLUDE_DIR NAMES unicode/utypes.h HINTS ${PC_ICU_INCLUDE_DIRS} ${PC_ICU_INCLUDEDIR})
+  # find_library(ICU_LIBRARY NAMES libicuuc cygicuuc cygicuuc32 icuuc HINTS ${PC_ICU_LIBRARY_DIRS} ${PC_ICU_LIBDIR})
 
-  include_directories(${ICU_INCLUDE_DIR})
+  # include_directories(${ICU_INCLUDE_DIR})
 
   list(APPEND project_headers ${PROJECT_SOURCE_DIR}/src/hb-icu.h)
 
-  list(APPEND THIRD_PARTY_LIBS ${ICU_LIBRARY})
+  # list(APPEND THIRD_PARTY_LIBS ${ICU_LIBRARY})
 
-  mark_as_advanced(ICU_INCLUDE_DIR ICU_LIBRARY)
+  # mark_as_advanced(ICU_INCLUDE_DIR ICU_LIBRARY)
 endif ()
 
 if (APPLE AND HB_HAVE_CORETEXT)
